substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/classinitialization/ClassInitializationFeature.java

```diff
@@ -368,17 +369,22 @@ static boolean declaresDefaultMethods(ResolvedJavaType type) {
             /* Only interfaces can declare default methods. */
             return false;
         }
-        /*
-         * We call getDeclaredMethods() directly on the wrapped type. We avoid calling it on the
-         * AnalysisType because it resolves all the methods in the AnalysisUniverse.
-         */
-        for (ResolvedJavaMethod method : Inflation.toWrappedType(type).getDeclaredMethods()) {
-            if (method.isDefault()) {
-                assert !Modifier.isStatic(method.getModifiers()) : "Default method that is static?";
-                return true;
+
+        try {
+            /*
+             * We call getDeclaredMethods() directly on the wrapped type. We avoid calling it on the
+             * AnalysisType because it resolves all the methods in the AnalysisUniverse.
+             */
+            for (ResolvedJavaMethod method : Inflation.toWrappedType(type).getDeclaredMethods()) {
+                if (method.isDefault()) {
+                    assert !Modifier.isStatic(method.getModifiers()) : "Default method that is static?";
+                    return true;
+                }
             }
+            return false;
+        } catch (NoClassDefFoundError e) {
+            throw new GraalError(e, "Unable to resolve all classes in class: %s", type.toClassName());
```

Comment:
Is there an action we could suggest to the user that gets this error?

Also, should we re-throw if `AllowIncompleteClasspath` is enabled? Maybe we should just ignore in that case and return false?

Reply:
@vjovanov was the intention here to return `false` when `AllowIncompleteClasspath` is *enabled*? This got merged with the condition flipped: 
```   
if (!NativeImageOptions.AllowIncompleteClasspath.getValue()) {
       return false;
}
```
Either way  returning `false` (assume that the interface does not have `default` methods) doesn't seem like a good idea. What are the implications of this on the class initialization logic that depends on it?

Reply:
@cstancu I was just looking at this and noticed the same thing. Indeed it seems the condition is flipped.

@vjovanov @cstancu Shall we need the check for `AllowIncompleteClasspath` assuming it's fixed to be:

```
if (NativeImageOptions.AllowIncompleteClasspath.getValue()) {
       return false;
}
```

Reply:
I've created https://github.com/oracle/graal/issues/2556 to fix this.